### PR TITLE
 Add integration test for per-resource storage encoding.

### PR DESCRIPTION
**Pull request by benluddy**
#### What type of PR is this?

/kind cleanup
/sig api-machinery

#### What this PR does / why we need it:

Verifies that each resource is stored with the expected encoding.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by benluddy** [comment]
/cc @deads2k 

/hold

Written on top of the unmerged https://github.com/kubernetes/kubernetes/pull/128539.

File: test/integration/apiserver/apiserver_test.go [hunk 1]
```diff
@@ -3352,3 +3360,132 @@ func assertManagedFields(t *testing.T, obj *unstructured.Unstructured) {
 func int32Ptr(i int32) *int32 {
 	return &i
 }
+
+// TestDefaultStorageEncoding verifies the storage encoding of each resource.
```
Comment (deads2k) [inline test/integration/apiserver/apiserver_test.go:3364]: expand on details.  All built-ins should use protobuf unless we've had an accident.  Looks like have two existing accidents.

**Comment by pacoxu** [comment]
/hold

The failure in pull-kubernetes-integration seems to be not a flake. Would you take a look?

**Comment by pacoxu** [comment]
```

=== Failed
=== FAIL: test/integration/apiserver  (0.00s)
I1106 01:00:32.941262  111887 etcd.go:73] etcd already running at http://127.0.0.1:2379
PASS
E1106 01:05:32.628124  111887 etcd.go:236] "EtcdMain goroutine check" err=<
	found unexpected goroutines:
	[Goroutine 116353 in state IO wait, 2 minutes, with internal/poll.runtime_pollWait on top of the stack:
	internal/poll.runtime_pollWait(0x7f366250b808, 0x72)
		/usr/local/go/src/runtime/netpoll.go:351 +0x85
	internal/poll.(*pollDesc).wait(0xc00b395a00?, 0xc00b502000?, 0x0)
		/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
	internal/poll.(*pollDesc).waitRead(...)
		/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
	internal/poll.(*FD).Read(0xc00b395a00, {0xc00b502000, 0x8000, 0x8000})
		/usr/local/go/src/internal/poll/fd_unix.go:165 +0x27a
	net.(*netFD).Read(0xc00b395a00, {0xc00b502000?, 0x1060100000000?, 0x8?})
		/usr/local/go/src/net/fd_posix.go:55 +0x25
	net.(*conn).Read(0xc005168f78, {0xc00b502000?, 0x801010601?, 0xc000000000?})
		/usr/local/go/src/net/net.go:189 +0x45
	bufio.(*Reader).Read(0xc00b416c60, {0xc0079bde00, 0x9, 0xc007cad808?})
		/usr/local/go/src/bufio/bufio.go:241 +0x197
	io.ReadAtLeast({0x5821cc0, 0xc00b416c60}, {0xc0079bde00, 0x9, 0x9}, 0x9)
		/usr/local/go/src/io/io.go:335 +0x90
	io.ReadFull(...)
		/usr/local/go/src/io/io.go:354
	golang.org/x/net/http2.readFrameHeader({0xc0079bde00, 0x9, 0xc004946540?}, {0x5821cc0?, 0xc00b416c60?})
		/home/prow/go/src/k8s.io/kubernetes/vendor/golang.org/x/net/http2/frame.go:237 +0x65
	golang.org/x/net/http2.(*Framer).ReadFrame(0xc0079bddc0)
		/home/prow/go/src/k8s.io/kubernetes/vendor/golang.org/x/net/http2/frame.go:501 +0x85
	google.golang.org/grpc/internal/transport.(*http2Client).reader(0xc002bd2fc8, 0xc004d61110)
		/home/prow/go/src/k8s.io/kubernetes/vendor/google.golang.org/grpc/internal/transport/http2_client.go:1620 +0x226
	created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 116380
		/home/prow/go/src/k8s.io/kubernetes/vendor/google.golang.org/grpc/internal/transport/http2_client.go:409 +0x1e53
	 Goroutine 116378 in state select, 2 minutes, with google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run on top of the stack:
	google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc00356c9f0, {0x5862e30, 0xc00b4d6230})
		/home/prow/go/src/k8s.io/kubernetes/vendor/google.golang.org/grpc/internal/grpcsync/callback_serializer.go:76 +0x10c
	created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 108653
		/home/prow/go/src/k8s.io/kubernetes/vendor/google.golang.org/grpc/internal/grpcsync/callback_serializer.go:52 +0x11a
	 Goroutine 116377 in state select, 2 minutes, with google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run on top of the stack:
	google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc00356c9c0, {0x5862e30, 0xc00b4d61e0})
		/home/prow/go/src/k8s.io/kubernetes/vendor/google.golang.org/grpc/internal/grpcsync/callback_serializer.go:76 +0x10c
	created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 108653
		/home/prow/go/src/k8s.io/kubernetes/vendor/google.golang.org/grpc/internal/grpcsync/callback_serializer.go:52 +0x11a
	 Goroutine 116379 in state select, 2 minutes, with google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run on top of the stack:
	google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc00356ca20, {0x5862e30, 0xc00b4d6280})
		/home/prow/go/src/k8s.io/kubernetes/vendor/google.golang.org/grpc/internal/grpcsync/callback_serializer.go:76 +0x10c
	created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 108653
		/home/prow/go/src/k8s.io/kubernetes/vendor/google.golang.org/grpc/internal/grpcsync/callback_serializer.go:52 +0x11a
	 Goroutine 116381 in state select, 2 minutes, with google.golang.org/grpc/internal/transport.(*controlBuffer).get on top of the stack:
	google.golang.org/grpc/internal/transport.(*controlBuffer).get(0xc00b3d54f0, 0x1)
		/home/prow/go/src/k8s.io/kubernetes/vendor/google.golang.org/grpc/internal/transport/controlbuf.go:418 +0x113
	google.golang.org/grpc/internal/transport.(*loopyWriter).run(0xc004d61180)
		/home/prow/go/src/k8s.io/kubernetes/vendor/google.golang.org/grpc/internal/transport/controlbuf.go:552 +0x7b
	google.golang.org/grpc/internal/transport.newHTTP2Client.func6()
		/home/prow/go/src/k8s.io/kubernetes/vendor/google.golang.org/grpc/internal/transport/http2_client.go:467 +0xbb
	created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 116380
		/home/prow/go/src/k8s.io/kubernetes/vendor/google.golang.org/grpc/internal/transport/http2_client.go:465 +0x243b
	]
 >
FAIL	k8s.io/kubernetes/test/integration/apiserver	299.819s
```

/test pull-kubernetes-integration

**Comment by benluddy** [comment]
/hold cancel

Seems to be fixed now.

**Pull request merged**